federatedscope/vertical_fl/xgb_base/worker/evaluation_wrapper.py

```diff
@@ -0,0 +1,213 @@
+import types
+import logging
+import numpy as np
+
+from federatedscope.vertical_fl.loss.utils import get_vertical_loss
+from federatedscope.core.message import Message
+
+logger = logging.getLogger(__name__)
+
+
+def wrap_client_for_evaluation(client):
+    def eval(self, tree_num):
+        self.criterion = get_vertical_loss(self._cfg.criterion.type)
+        if self.test_x is None:
+            self.test_x, self.test_y = self._fetch_test_data()
+            self.test_result = np.zeros(self.test_x.shape[0])
+        self.model[tree_num][0].indicator = np.ones(self.test_x.shape[0])
+        self._test_for_node(tree_num, node_num=0)
+
+    def _fetch_test_data(self):
+        test_x = self.data['test']['x']
+        test_y = self.data['test']['y'] if 'y' in self.data['test'] else None
+
+        return test_x, test_y
+
+    def _feedback_eval_metrics(self):
+        test_loss = self.criterion.get_loss(self.test_y, self.test_result)
+        metrics = self.criterion.get_metric(self.test_y, self.test_result)
+        modified_metrics = dict()
+        for key in metrics.keys():
+            if 'test' not in key:
+                modified_metrics['test_' + key] = metrics[key]
+            else:
+                modified_metrics[key] = metrics[key]
+        modified_metrics.update({
+            'test_loss': test_loss,
+            'test_total': len(self.test_y)
+        })
+
+        self.comm_manager.send(
+            Message(msg_type='eval_metric',
+                    sender=self.ID,
+                    state=self.state,
+                    receiver=[self.server_id],
+                    content=modified_metrics))
+        self.comm_manager.send(
+            Message(msg_type='feature_importance',
+                    sender=self.ID,
+                    state=self.state,
+                    receiver=[self.server_id],
+                    content=self.feature_importance))
+        self.comm_manager.send(
+            Message(msg_type='ask_for_feature_importance',
+                    sender=self.ID,
+                    state=self.state,
+                    receiver=[
+                        each
+                        for each in list(self.comm_manager.neighbors.keys())
+                        if each != self.server_id
```

Comment:
Here should add the condition that "and each != self.ID", otherwise label owner would send it twice.